genkit-tools/common/src/manager/manager.ts

```diff
@@ -198,9 +198,25 @@ export class RuntimeManager {
         rejecter = reject;
       });
       stream.on('end', () => {
-        const actionResponse = RunActionResponseSchema.parse(
-          JSON.parse(buffer)
-        );
+        const parsedBuffer = JSON.parse(buffer);
+        if (parsedBuffer.error) {
+          const err = new GenkitToolsError(
+            `Error running action key='${input.key}'.`
+          );
+          // massage the error into a shape dev ui expects
```

Comment:
Should we/do we have a common type for this? Feels like we should given we share types between CLI/Dev UI.

Reply:
Done.